### ISSUE TITLE
misplaced error message when using @as

### ISSUE THREAD

**Issue by taikoo** [ziglang/zig#16857]
### Zig Version

0.11.0

### Steps to Reproduce and Observed Output

```zig
const std = @import("std");

pub fn main() void {
    var a: u64 = 1567;

    var b: u16 = @as(u16, a);

    std.debug.print("b: {}\n", .{b});
}
```

```
$ zig build-exe casting.zig 
casting.zig:3:1: error: expected type 'u16', found 'u64'
pub fn main() void {
^~~
casting.zig:3:1: note: unsigned 16-bit int cannot represent all possible unsigned 64-bit values
```

### Expected Output

error message should point to:
```zig
var b: u16 = @as(u16, a);
                      ^
```

**Comment by xdBronch** [comment]
this has been fixed on master, im on `0.12.0-dev.89+913511557` and it correctly points to the `@as(u16, a)` line

**Comment by andrewrk** [comment]
will be included in the upcoming 0.11.1 bug fix release. closing as duplicate, solved by #16787
